soroban-env-host/src/storage.rs

```diff
@@ -19,6 +19,20 @@ use crate::{host::metered_map::MeteredOrdMap, HostError};
 
 pub type FootprintMap = MeteredOrdMap<Rc<LedgerKey>, AccessType, Budget>;
 pub type StorageMap = MeteredOrdMap<Rc<LedgerKey>, Option<Rc<LedgerEntry>>, Budget>;
+#[derive(Clone)]
+pub struct InstanceStorageMap {
+    pub map: MeteredOrdMap<Val, Val, Host>,
```

Comment:
Can this be changed to a `Budget` instead of `Host`? As far as I can tell this doesn't need anything else from the Host besides the budget. This way it can be used outside of the host (e.g. setting up a storage without the host).

Reply:
I don't think so - this stores `Val` and `Val` is meaningless outside of the host (that's why we don't have comparison implemented for `<Val, Budget>`. Moreover, this is deeply internal data structure and it shouldn't be used oustside of the host (I'll actually switch it to pub(crate)).

Reply:
Ok, makes sense.